### PR TITLE
Merge stackdriver-core-dependencies feature branch to master

### PR DESCRIPTION
**Pull request by dazuma**
Brings feature work that depends on stackdriver-core into master in preparation to release the following gems:

* stackdriver-core v0.21
* google-cloud-logging v0.23.1 (dependent on stackdriver-core)
* google-cloud-trace v0.21 (dependent on stackdriver-core)

This branch includes the following PRs:

* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1136 "Update google-cloud-logging to use trace context tracking from core"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1137 "Trace instrumentation"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1145 "Basic acceptance tests for google-cloud-trace"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1148 "Logger object understands all attributes supported by the stdlib Logger"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1150 "Ensure AsyncWriter threads finish on VM exit"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1151 "Improve Logging::Railtie activation warning messages"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1152 "Update and clarify docs for google-cloud-trace"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1153 "Remove orphaned span hack from middleware, and fix conversion of orphaned spans from protos"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1157 "Trace sampler interface is now just a Proc"
* https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1159 "Rename to_ and from_proto to to_ and from_grpc for trace models"

### CODE REVIEW

File: google-cloud-trace/docs/toc.json [hunk 1]
```diff
@@ -1,8 +1,71 @@
...
   "services": [
     {
-      "title": "TraceServiceApi",
+      "title": "Google::Cloud",
+      "type": "google/cloud"
+    },
+    {
+      "title": "Google::Cloud::Trace",
+      "type": "google/cloud/trace",
+      "nav": [
```
Comment (blowmage) [inline google-cloud-trace/docs/toc.json]: @quartzmo What do you think of the navigation? Will the app handle this many nodes? We've been trying to keep the number of sub-nav nodes to ~5.

Comment (quartzmo) [inline google-cloud-trace/docs/toc.json]: The site app supports it for sure, but I can't comment on the usability.

Comment (dazuma) [inline google-cloud-trace/docs/toc.json]: In general, for libraries that have a number of classes, have we been including only a subset of them in the nav? How have we generally been keeping the nav small?

Comment (quartzmo) [inline google-cloud-trace/docs/toc.json]: @dazuma Our tactic has been to make sure that "secondary" classes are accessible (linked) as return types or params. But the site is lacking both index and search, so I fully understand if you feel it is the best decision to just list them all in the nav.

Comment (dazuma) [inline google-cloud-trace/docs/toc.json]: I'm guessing ruby developers would expect to see essentially a class reference in a left nav (for which the current list of 10 classes should not be unusual or difficult to use.) Even if we also have a section of the docs that is more topical, the reference section is also needed. Especially given that content comes from the yardoc, I do think it's best to structure that part of the nav accordingly.

Comment (quartzmo) [inline google-cloud-trace/docs/toc.json]: Sounds good to me. I think the original concept was more of curated, guided on-boarding experience, but we should revisit that idea.

**Comment by dazuma** [comment]
appveyor you've had 24 hours...

**Comment by blowmage** [comment]
@dazuma Can you confirm the dependency on google-cloud-core 0.20.0 will work?

**Comment by dazuma** [comment]
@blowmage not sure what you mean? google-cloud-trace and google-cloud-logging both declare `"google-cloud-core", "~> 0.21.0"`. I think that probably should be `"~> 0.21.1"` to pick up the GAE Flex Beta 2 fixes. Is that what you're talking about?

**Comment by blowmage** [comment]
Yes, that was the question I posed yesterday.

**Comment by dazuma** [comment]
@blowmage Ah sorry, I missed that. Yes, lemme update the dependencies and reverify...

**Comment by blowmage** [comment]
Awesome. Thanks! I just noticed that I asked in a review and not a comment on the code. I blame GitHub's UI.

**Comment by dazuma** [comment]
@blowmage Updated google-cloud-core dependencies to `~> 0.21.1`. I still can't find the rest of your comments for some reason. Was there anything else I missed?

**Comment by blowmage** [comment]
Nope. That was it.

**Comment by blowmage** [comment]
Didn't mean to close this. I rebased the branch on master and it was closed.

**Pull request merged**